provider/identity_provider_resource.go

```diff
@@ -283,6 +310,22 @@ func (r *IdentityProviderResource) Create(ctx context.Context,
 			htpasswdBuilder.Password(state.HTPasswd.Password.Value)
 		}
 		builder.Htpasswd(htpasswdBuilder)
+    case state.Gitlab != nil:
+		builder.Type(cmv1.IdentityProviderType("GitlabIdentityProvider"))
+		gitlabBuilder := cmv1.NewGitlabIdentityProvider()
+		if !state.Gitlab.CA.Null {
+			gitlabBuilder.CA(state.Gitlab.CA.Value)
+		}
+		if !state.Gitlab.ClientID.Null {
+			gitlabBuilder.ClientID(state.Gitlab.ClientID.Value)
+		}
+		if !state.Gitlab.ClientSecret.Null {
+			gitlabBuilder.ClientSecret(state.Gitlab.ClientSecret.Value)
+		}
+		if !state.Gitlab.URL.Null {
+			gitlabBuilder.URL(state.Gitlab.URL.Value)
```

Comment:
I saw that ROSA cli also validate the HostURL  
https://github.com/openshift/rosa/blob/master/cmd/create/idp/gitlab.go#L159

Can you please add that validation before setting the value of `gitlabBuilder.URL`?

Reply:
Struggled with this part, let me know if validation added is acceptable.

Reply:
@oriAdler  please review the validation URL also